### PR TITLE
[exporter/datadog] Use v0.3.0 of opentelemetry-mapping-go for logs tr…

### PR DESCRIPTION
**Pull request by IbraheemA**
…anslation

**Description:** 
translator.go was copied into the `DataDog/opentelemetry-mapping-go` repo in a previous PR, which is now available in the v0.3.0 release of that repo. Update the dependency version and replace imports of `translator.go` with the `opentelemetry-mapping-go` equivalent.

**Testing:** Tested with `logs-appender` example that sends logs and traces.

**Documentation:**

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
Please resolve the merge conflicts and add a changelog

File: .chloggen/otel-529-use-mapping-module.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: exporter/datadog
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Use v0.3.0 of opentelemetry-mapping-go for logs translation
```
Comment (songy23) [inline .chloggen/otel-529-use-mapping-module.yaml:12]:
```suggestion
note: Upgrade opentelemetry-mapping-go to v0.3.0 and use pkg/otlp/logs for logs translation
```

File: receiver/iisreceiver/integration_test.go [hunk 2]
```diff
@@ -10,9 +10,10 @@ import (
 	"testing"
 	"time"
 
+	"go.opentelemetry.io/collector/component"
```
Comment (songy23) [inline receiver/iisreceiver/integration_test.go:13]: This seems irrelevant to your change?

Comment (IbraheemA) [reply receiver/iisreceiver/integration_test.go:13]: `tidy` made this change, should I remove it?

**Review by songy23 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
